x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -118,6 +122,8 @@ export const WorkspacePanel = React.memo(function WorkspacePanel(props: Workspac
   );
 });
 
+let expressionToRender: null | undefined | string;
```

Comment:
Please do not use global state like that - it will cause very weird errors as soon as there are two workspace panels active for some reason. It looks like you want to avoid re-renders on changing, you can achieve this using the `useRef` hook which allows you to store a mutable variable which is not triggering re-evaluation on it's own (like a class property in class based react components): https://reactjs.org/docs/hooks-reference.html#useref

Reply:
Awesome 👍. I felt that this wasn't the true React way but didn't know what was.